hubverse_annotator/app.py

```diff
@@ -0,0 +1,210 @@
+            elif ext == ".csv":
+                smhub_table = pl.read_csv(uploaded_file)
+            else:
+                raise ValueError(f"Unsupported file type: {ext}")
+        except ValueError as e:
+            st.error(str(e))
+            st.stop()
+        st.success(f"Loaded {uploaded_file.name} ({ext}).")
+        logger.info(f"Uploaded file:\n{uploaded_file.name}")
+        logger.info(f"Contents\n:{smhub_table}")
+        # locations in the hubverse table
+        smhub_loc_codes = smhub_table["location"].unique().to_list()
+        loc_lookup = forecasttools.location_lookup(
+            location_vector=smhub_loc_codes, location_format="hubverse"
+        )
+        locs_available = loc_lookup["long_name"].to_list()
+        # two-column layout for reference date and location
+        col1, col2 = st.columns(2)
+        with col1:
+            ref_dates = (
+                smhub_table["reference_date"]
+                .unique()
+                .sort()
+                .dt.strftime("%Y-%m-%d")
+                .to_list()
+            )
+            selected_ref_date = st.selectbox(
+                "Reference Date",
+                options=ref_dates,
+                key="ref_date_selection",
+            )
+        with col2:
+            location = st.selectbox(
+                "Location",
+                options=locs_available,
+            )
+        # filter to location before filtering to model
+        two_num_loc_abbr = (
+            loc_lookup.filter(pl.col("long_name") == location)
+            .get_column("short_name")
+            .item()
+        )
+        two_letter_loc_abbr = (
+            loc_lookup.filter(pl.col("long_name") == location)
+            .get_column("location_code")
+            .item()
+        )
+        smhubt_by_loc = smhub_table.filter(
+            pl.col("location") == two_num_loc_abbr,
+        )
+        # models and targets available
+        models_available = smhubt_by_loc["model"].unique().to_list()
+        selected_models = st.multiselect(
+            "Model(s)",
+            options=models_available,
+            key="model_selection",
+        )
+        targets_available = smhubt_by_loc["target"].unique().to_list()
+        selected_target = st.selectbox(
+            "Target(s)", options=targets_available, key="target_selection"
+        )
+        # filter hubverse table by selected models and target
+        smhubt_to_plot = smhubt_by_loc.filter(
+            pl.col("model").is_in(selected_models),
+            pl.col("target") == selected_target,
+        )
+
+        st.markdown(f"## Forecasts For: {two_num_loc_abbr}")
+        st.markdown(f"## Reference Date: {selected_ref_date}")
+        # plotting of the selected model, target, location, and reference date
+        if smhubt_to_plot.is_empty():
+            st.warning(
+                f"No models selected or no forecasts available for the "
+                f"location: {location}"
+            )
```

Comment:
I think just revert to the error you had before

```suggestion
            st.warning(
                f"No forecasts available for the combination\nReference "
                f"Date: {selected_ref_date}\nModels: {selected_models}\n"
                f"Target: {selected_target}"
            )

```